handsontable/src/plugins/manualColumnMove/manualColumnMove.js

```diff
@@ -406,53 +406,59 @@ export class ManualColumnMove extends BasePlugin {
     const lastVisible = this.hot.view.wt.wtTable.getLastVisibleColumn();
     const wtTable = this.hot.view.wt.wtTable;
     const scrollableElement = this.hot.view.wt.wtOverlays.scrollableElement;
-    const scrollLeft = typeof scrollableElement.scrollX === 'number' ?
+    const scrollStart = typeof scrollableElement.scrollX === 'number' ?
       scrollableElement.scrollX : scrollableElement.scrollLeft;
-    let tdOffsetLeft = this.hot.view.THEAD.offsetLeft + this.getColumnsWidth(0, priv.coords - 1);
-    const mouseOffsetLeft = priv.target.eventPageX - (priv.rootElementOffset - (scrollableElement.scrollX === void 0 ? scrollLeft : 0)); // eslint-disable-line max-len
+    let tdOffsetStart = this.hot.view.THEAD.offsetLeft + this.getColumnsWidth(0, priv.hoveredColumn - 1);
     const hiderWidth = wtTable.hider.offsetWidth;
     const tbodyOffsetLeft = wtTable.TBODY.offsetLeft;
-    const backlightElemMarginLeft = this.backlight.getOffset().left;
+    const backlightElemMarginStart = this.backlight.getOffset().start;
     const backlightElemWidth = this.backlight.getSize().width;
     let rowHeaderWidth = 0;
+    let mouseOffsetStart = 0;
 
-    if ((priv.rootElementOffset + wtTable.holder.offsetWidth + scrollLeft) < priv.target.eventPageX) {
-      if (priv.coords < priv.countCols) {
-        priv.coords += 1;
-      }
+    if (this.hot.isRtl()) {
+      const rootWindow = this.hot.rootWindow;
+      const containerWidth = outerWidth(this.hot.rootElement);
+      const gridMostRightPos = rootWindow.innerWidth - priv.rootElementOffset - containerWidth;
+
+      mouseOffsetStart = rootWindow.innerWidth - priv.target.eventPageX - gridMostRightPos -
+        (scrollableElement.scrollX === void 0 ? scrollStart : 0);
+    } else {
+      mouseOffsetStart = priv.target.eventPageX -
+        (priv.rootElementOffset - (scrollableElement.scrollX === void 0 ? scrollStart : 0));
     }
 
     if (priv.hasRowHeaders) {
       rowHeaderWidth = this.hot.view.wt.wtOverlays.inlineStartOverlay.clone.wtTable.getColumnHeader(-1).offsetWidth;
     }
-    if (this.isFixedColumnsStart(priv.coords)) {
-      tdOffsetLeft += scrollLeft;
+    if (this.isFixedColumnsStart(priv.hoveredColumn)) {
```

Comment:
An empty line between the  `if`s would be nice, as well as an empty line after the second `if`. (I know it's nitpicking and that most of the code here is not new, so I leave it up to you)